### PR TITLE
feat(codeaction): add UnnecessaryGString quick fix

### PR DESCRIPTION
**Pull request by albertocavalcante**
## Summary
- Add quick fix for `UnnecessaryGString` CodeNarc rule that converts double-quoted strings to single-quoted strings
- Fix diagnostic range positioning bug (was "padded left" due to incorrect quote matching in message extraction)

## Changes
- **RuleRangeCalculator.kt**: Fix `extractQuotedValue()` to match single quotes from CodeNarc message (the message format is `'content'` not `"content"`)
- **FixHandlerRegistry.kt**: Add `fixUnnecessaryGString` handler that converts `"string"` to `'string'`, returning null for strings containing single quotes

## Test Plan
- [x] 11 range calculation tests in `CodeNarcDiagnosticFixTest`
- [x] 10 handler tests (including property-based) in `ConventionFixHandlersTest`
- [x] 5 integration tests in `CodeActionProviderTest`
- [x] All existing tests pass
- [x] Lint passes

---

> [!NOTE]
> Adds a safe quick fix for unnecessary GStrings and tightens range calculation for those diagnostics.
> 
> - **Code actions:** Registers `UnnecessaryGString` with `fixUnnecessaryGString` to convert `"..."` to `'...'` when safe (skips content with single quotes or escaped double quotes)
> - **Diagnostics:** Updates `RuleRangeCalculator.extractQuotedValue()` to prefer single-quoted values in messages (fallback to double-quoted), improving `UnnecessaryGString` column accuracy with robust fallback to first quoted span
> - **Tests:** Adds comprehensive unit, property-based, and integration tests covering range positioning and code-action edits across edge cases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 99850b90d05edd973fb2e934e128778d89835481. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

## Summary by CodeRabbit

* **New Features**
  * Added "Use single-quoted string" quick fix for unnecessary GString diagnostics, automatically converting double-quoted strings to single-quoted equivalents when safe.

* **Improvements**
  * Enhanced string content parsing to handle both single and double-quoted values.
  * Improved diagnostic range calculations for more accurate problem identification across various string patterns and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: groovy-lsp/src/main/kotlin/com/github/albertocavalcante/groovylsp/providers/codeaction/FixHandlerRegistry.kt [hunk 1]
```diff
@@ -625,3 +629,73 @@ private fun fixUnnecessaryDotClass(context: FixContext): TextEdit? {
...
+    }
+
+    // Extract the content (without the surrounding quotes)
+    val content = gstring.substring(1, gstring.length - 1)
+
+    // If the content contains single quotes, we cannot convert
+    // (it would break the syntax: 'it's' is invalid)
+    if (content.contains("'")) {
+        return null
+    }
```
[the remark this answers is not in the document]
Comment (albertocavalcante) [reply groovy-lsp/src/main/kotlin/com/github/albertocavalcante/groovylsp/providers/codeaction/FixHandlerRegistry.kt]: Fixed in 99850b90. Handler now returns null for strings containing escaped double quotes.
